components/vision.py

```diff
@@ -7,9 +7,33 @@ def __init__(self) -> None:
         self.nt = NetworkTables
         self.table = self.nt.getTable("/vision")
         self.entry = self.table.getEntry("data")
+        self.lastTime = 0
+        self.repeats = 0
+        self.data = [None, None, None]
 
     def get_vision_data(self) -> Tuple[float, float, float]:
         """Returns a tuple containing the distance (metres), angle (radians), and timestamp (time.monotonic)
-        If it can't get info, it returns [None, None, None]
+        If it can't get the entry, it returns [None, None, None]
+
+        If the pi vision cant see the target it will not send to network tables, meaning this will continue
+        to return the most recently seen target position and the time it was seen
         """
-        return self.entry.getDoubleArray([None, None, None])
+        if self.entry.exists():
```

Comment:
... isn't this saying exactly the same thing twice?

Reply:
yes, i didn't realise they were doing the same thing and have removed the entry.exists check